src/components/RiddleDetail.tsx

```diff
@@ -105,15 +107,13 @@ export const RiddleDetail = ({ isCreatorView, riddleDetail }: Props) => {
 				}}
 			>
 				<Button
-					type="submit"
 					variant="contained"
 					sx={{ backgroundColor: 'primary.light', flex: 1, maxWidth: '200px' }}
 					onClick={() => navigate(-1)}
```

Comment:
Tady vznikla taková zajímavá chyba. Když jdu z detailu hádanky na její řešení a tam kliknu na "Back to detail" a pak v detailu kliknu na "Back", dostanu se zase zpátky na řešení. :D Opravím to.